src/vault/client/http.clj

```diff
@@ -213,17 +218,19 @@
 
 ;; ## HTTP Client Type
 
-;; - `:api-url`
+;; - `api-url`
 ;;   The base URL for the Vault API endpoint.
-;; - `:auth`
+;; - `http-opts`
+;;   Extra options to pass to `clj-http` requests.
+;; - `auth`
 ;;   An atom containing the authentication lease information, including the
 ;;   client token.
-;; - `:leases`
+;; - `leases`
 ;;   Local in-memory storage of secret leases.
-;; - `:lease-timer`
+;; - `lease-timer`
 ;;   Thread which periodically checks and renews leased secrets.
 (defrecord HTTPClient
-  [api-url auth leases lease-timer]
+  [api-url http-opts auth leases lease-timer]
```

Comment:
since this breaks older consumers of the api maybe leave the 4-arity and add a 5-arity with the opts as the last arg?